dispatch/queue.h

```diff
@@ -990,6 +990,54 @@ DISPATCH_EXPORT DISPATCH_PURE DISPATCH_WARN_RESULT DISPATCH_NOTHROW
 void *
 dispatch_get_specific(const void *key);
 
+#if __linux__
```

Comment:
Although I agree that this should not be in the public header, I do think it makes sense to make dedicated entry points for this like we also have on Windows. That means that we can change them independently from each other without having to worry about binary compatibility.